### PR TITLE
Clean up header dependency.

### PR DESCRIPTION
**Pull request by antiagainst**

### CODE REVIEW

File: source/opt/fold_spec_constant_op_and_composite_pass.cpp [hunk 1]
```diff
@@ -249,6 +247,20 @@ FoldSpecConstantOpAndCompositePass::FoldSpecConstantOpAndCompositePass()
...
+Pass::Status FoldSpecConstantOpAndCompositePass::Process(ir::Module* module) {
+  Initialize(module);
+  return ProcessImpl(module);
+}
+
+void FoldSpecConstantOpAndCompositePass::Initialize(ir::Module* module) {
+  type_mgr_.reset(new analysis::TypeManager(consumer(), *module));
+  def_use_mgr_.reset(new analysis::DefUseManager(consumer(), module));
+  for (const auto& id_def : def_use_mgr_->id_to_defs()) {
+    max_id_ = std::max(max_id_, id_def.first);
```
Comment (dneto0) [inline source/opt/fold_spec_constant_op_and_composite_pass.cpp]: std::Max needs algorithm.  Add it here, remove it from the .h file.

File: source/opt/fold_spec_constant_op_and_composite_pass.h [hunk 2]
```diff
@@ -35,22 +35,13 @@ class FoldSpecConstantOpAndCompositePass : public Pass {
   FoldSpecConstantOpAndCompositePass();
 
   const char* name() const override { return "fold-spec-const-op-composite"; }
-  Status Process(ir::Module* module) override {
-    Initialize(module);
-    return ProcessImpl(module);
-  };
+
+  Status Process(ir::Module* module) override;
```
Comment (dneto0) [inline source/opt/fold_spec_constant_op_and_composite_pass.h]: Suspect you don't need algorithm or memory anymore.  Add it to the corresponding .cpp file?

File: source/opt/fold_spec_constant_op_and_composite_pass.cpp (hunk 1 above)
Comment (antiagainst) [inline source/opt/fold_spec_constant_op_and_composite_pass.cpp]: Good catch!

File: source/opt/fold_spec_constant_op_and_composite_pass.h (hunk 2 above)
Comment (antiagainst) [inline source/opt/fold_spec_constant_op_and_composite_pass.h]: Right. I've moved some impl. code into `.cpp` so `<algorithm>` definitely need to be moved. For `<memory>`, we have `std::unique_ptr` in the header.

**Pull request merged**